### PR TITLE
Feature/m0127 cc 629 tutorial non blocking sni

### PR DESCRIPTION
**Pull request by acolleux-microej**

### CODE REVIEW

**Review by jerome-leroux — CHANGES_REQUESTED** [review]

File: Tutorials/index.rst [hunk 1]
```diff
@@ -16,10 +16,11 @@ Tutorials
    tutorialExploreDataSerializationFormats
    tutorialInstrumentJavaCodeForLogging
    tutorialRunATestSuiteOnDevice
+   tutorialSNINonBlockingCall
```
Comment (frivieremicroej) [inline Tutorials/index.rst:19]:
```suggestion
   tutorialImplementABlockingNativeMethodWithSNI
```

File: Tutorials/tutorialSNINonBlockingCall.rst [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+Requirements
+============
+
+A MicroEJ Platform with (at least):
+
+- EDC-1.3.
+- SNI-1.4.
+
+Example code
```
Comment (frivieremicroej) [inline Tutorials/tutorialSNINonBlockingCall.rst:45]: Please review titles everywhere
```suggestion
Example Code
```

File: Tutorials/tutorialSNINonBlockingCall.rst [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+In this example, the execution of the ``waitButton()`` native method will block until a button is pressed. 
+In other words, while ``Java_example_NativeCCallExample_waitButton()`` has not returned, no other Java thread can
+be scheduled.
+
+This is because the native function is called in the same RTOS/OS task as the Java application.
+
+This schematic explains what is going on:
+
+.. image:: ./images/tuto_sni_non_blocking_call_blocking_case.PNG
```
Comment (frivieremicroej) [inline Tutorials/tutorialSNINonBlockingCall.rst:110]: Please submit all schematics to design review (especially colors review: light-orange background colors/ yellow font...)

**Review by frivieremicroej — CHANGES_REQUESTED** [review]

File: Tutorials/index.rst (hunk 1 above)
Comment (jerome-leroux) [reply Tutorials/index.rst]: done

File: Tutorials/tutorialSNINonBlockingCall.rst (hunk 2 above)
Comment (jerome-leroux) [reply Tutorials/tutorialSNINonBlockingCall.rst]: done

File: Tutorials/tutorialSNINonBlockingCall.rst (hunk 3 above)
Comment (jerome-leroux) [reply Tutorials/tutorialSNINonBlockingCall.rst]: @acolleux-microej: can you do that?

Comment (acolleux-microej) [reply Tutorials/tutorialSNINonBlockingCall.rst]: @jerome-leroux @frivieremicroej I updated the schematics

File: Tutorials/tutorialImplementABlockingNativeMethodWithSNI.rst [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+================================================
+
+This tutorial describes the good practices to follow when implementing a blocking native method in C.
+A native method is a method declared in Java with the ``native`` keyword and implemented in C using the
+:ref:`Simple Native Interface (SNI) <sni>`.
+
+Intended Audience
+=================
+
+The audience for this document is Platform developers who want to implement abstraction layer interfaces.
```
Comment (frivieremicroej) [inline Tutorials/tutorialImplementABlockingNativeMethodWithSNI.rst:14]: please perform a whole pass on glossary terms
```suggestion
The audience for this document is Platform developers who want to implement Abstraction Layer interfaces.
```

File: Tutorials/tutorialImplementABlockingNativeMethodWithSNI.rst [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+Prerequisites
+=============
+
+The following document assumes the reader already has a setup ready to run a MicroEJ application on a target device.
```
Comment (frivieremicroej) [inline Tutorials/tutorialImplementABlockingNativeMethodWithSNI.rst:19]:
```suggestion
The following document assumes the reader already has a setup ready to run a MicroEJ Standalone Application on a target device.
```

File: Tutorials/tutorialImplementABlockingNativeMethodWithSNI.rst [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+A MicroEJ Platform with (at least):
+
+- EDC-1.3.
+- SNI-1.4.
+
+Example Code
+============
+
+Let's start with a MicroEJ Application that calls a blocking Java method implemented in C.
```
Comment (frivieremicroej) [inline Tutorials/tutorialImplementABlockingNativeMethodWithSNI.rst:48]:
```suggestion
Let's start with a MicroEJ Standalone Application that calls a blocking Java method implemented in C.
```

**Review by frivieremicroej — APPROVED** [review]
approved once glossary terms are commited

**Pull request merged**